src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -192,25 +194,32 @@ public String  MakeGUID() {
     public void BeginConnectPeer(String toPeerId, ConnectStatusCallback connectStatusCallback) {
         ConnectStatusCallback tmpCallback = connectStatusCallback;
 
+        // needed since mBtToRequestSocket can be disconnected from other thread
+        BtToRequestSocket tmp = mBtToRequestSocket;
+
         if(tmpCallback == null) {
             //nothing we should do, since we can not update progress
             print_debug("BeginConnectPeer callback is NULL !!!!!!");
-        }else if(mBtToRequestSocket != null) {
-            tmpCallback.ConnectionStatusUpdate("Already connected to " + mBtToRequestSocket.GetPeerId(),-1);
+        }else if(tmp != null) {
+            tmpCallback.ConnectionStatusUpdate("Already connected to " + tmp.GetPeerId(),-1);
         }else {
```

Comment:
@DrJukka I would suggest that the code would be easier to read if you just stuck a return after line 204. That would allow you to get rid of the next else and indent all of that code one tab space to the left. It's easier to read code that isn't deeply nested.